### PR TITLE
interfaces: add kernel-crypto-api interface

### PR DESCRIPTION
**Pull request by jdstrand**
interfaces: add kernel-crypto-api interface

The kernel crypto API is designed to be used by any process (ie, using
it requires no special privileges). Since it provides a kernel surface
(which internally may or may not utilize Hardware Security Modules
(HSMs) and has a CVE history, separate into a separate interface and
manually connect for now. It is easy enough for VPN software or network
security software (eg k8s/cilium) to request auto-connection.

References:
- https://www.kernel.org/doc/html/latest/crypto/userspace-if.html
- https://forum.snapcraft.io/t/denied-family-alg-sock-type-seqpacket/15832

### CODE REVIEW

**Comment by jdstrand** [comment]
> LGTM, thanks for this. PS the name kernel-crypto-api sounds ok to me, as per your comments on the forum I don't think `-control` or `-observe` is appropriate here.

Yeah, this is neither observe (we are using a service, not observing a value) and it isn't control (we aren't changing anything). The kernel crypto API is a service that the kernel provides that people can just use for crypto operations, typically to take advantage of HSMs (but not limited to that). Note that `-api` is not intended to be a new suffix to indicate a new category; I chose `kernel-crypto-api` since that is how documentation and users refer to the service (eg, `kernel-crypto-api` as a name is IME analogous to `mir` or `docker`).

**Comment by jdstrand** [comment]
The failures are unrelated to this PR: google:ubuntu-16.04-64:tests/main/cgroup-tracking:root. I've merged from master but I didn't see anything that would fix that....

**Comment by anonymouse64** [comment]
> The failures are unrelated to this PR: google:ubuntu-16.04-64:tests/main/cgroup-tracking:root. I've merged from master but I didn't see anything that would fix that....

These are known, but unfortunately not understood yet so there is not a fix. The issues are transient however, so restarting should make it (temporarily) go away probably.

File: interfaces/builtin/kernel_crypto_api.go [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+// https://www.kernel.org/doc/html/latest/crypto/userspace-if.html
+const kernelCryptoApiSummary = `allows access to the Linux kernel crypto API`
```
Comment (pedronis) [inline interfaces/builtin/kernel_crypto_api.go:23]: s/Api/API/ everywhere in names

File: interfaces/builtin/kernel_crypto_api.go [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+// https://www.kernel.org/doc/html/latest/crypto/userspace-if.html
```
Comment (pedronis) [inline interfaces/builtin/kernel_crypto_api.go:22]: could you add a link to this as well:  https://www.kernel.org/doc/html/latest/crypto/intro.html

it's a bit of a clearer reference to why the name

File: interfaces/builtin/kernel_crypto_api.go (hunk 1 above)
Comment (jdstrand) [reply interfaces/builtin/kernel_crypto_api.go:23]: Done

File: interfaces/builtin/kernel_crypto_api.go (hunk 2 above)
Comment (jdstrand) [reply interfaces/builtin/kernel_crypto_api.go:22]: Done

**Comment by jdstrand** [comment]
@pedronis - thanks for the review. I addressed all of you comments.

**Comment by jdstrand** [comment]
FYI, the test failure seems unrelated. All 20.10 tests were aborted.

File: interfaces/builtin/kernel_crypto_api_test.go [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+  kernel-crypto-api:
+`
+
+func (s *kernelCryptoAPIInterfaceSuite) SetUpTest(c *C) {
+	s.plug, s.plugInfo = MockConnectedPlug(c, kernelCryptoAPIConsumerYaml, nil, "kernel-crypto-api")
+	s.coreSlot, s.coreSlotInfo = MockConnectedSlot(c, kernelCryptoAPICoreYaml, nil, "kernel-crypto-api")
+}
+
+func (s *kernelCryptoAPIInterfaceSuite) TearDownTest(c *C) {
+	dirs.SetRootDir("/")
```
Comment (mvo5) [inline interfaces/builtin/kernel_crypto_api_test.go:66]: (nitpick) This seems to be not changed by this test so we can probably remove TearDownTest()

Comment (jdstrand) [reply interfaces/builtin/kernel_crypto_api_test.go]: Done. Thanks!

**Pull request merged**